packages/template-retail-react-app/app/components/_app/index.test.js

```diff
@@ -82,4 +85,68 @@ describe('App', () => {
         expect(hreflangLinks.some((link) => hasGeneralLocale(link))).toBe(true)
         expect(hreflangLinks.some((link) => link.hrefLang === 'x-default')).toBe(true)
     })
+
+    test('App component updates the basket with correct currency and customer email', async () => {
+        // Test basket. _app will be manipulating this basket for this test
+        const basket = {
+            basketId: 'basket_id',
+            currency: 'CAD',
+            customerInfo: {
+                customerId: 'customer_id'
+            }
+        }
+
+        jest.spyOn(
+            jest.requireActual('@salesforce/retail-react-app/app/hooks/use-current-customer'),
+            'useCurrentCustomer'
+        ).mockImplementation(() => {
+            return {
+                data: {
+                    customerId: 'customer_id',
+                    email: 'email@test.com',
+                    isRegistered: true
+                }
+            }
+        })
+
+        jest.spyOn(
+            jest.requireActual('@salesforce/retail-react-app/app/hooks/use-current-basket'),
+            'useCurrentBasket'
+        ).mockImplementation(() => {
+            return {
+                data: basket,
+                derivedData: {
+                    hasBasket: true,
+                    totalItems: 0
+                }
+            }
+        })
+
+        global.server.use(
+            // mock updating basket currency
+            rest.patch('*/baskets/:basketId', (req, res, ctx) => {
+                basket.currency = 'GBP'
+                return res(ctx.json(basket))
+            }),
+            // mock adding guest email to basket
+            rest.put('*/baskets/:basketId/customer', (req, res, ctx) => {
+                basket.customerInfo.email = 'customer@test.com'
+                return res(ctx.json(basket))
+            })
+        )
+
+        useMultiSite.mockImplementation(() => resultUseMultiSite)
+        await act(() =>
```

Comment:
We probably don't need act here since `render`  already comes with act 🤔

Reply:
The act is needed there as I get the `not wrapped in act` error without adding it there. I think it may be because what we're really waiting on is the useEffect that happens after the initial render.

Reply:
Instead of using act, we should wait for the basket to get changed
```
 await waitFor(() => {
            expect(basket.currency).toBe('GBP')
            expect(basket.customerInfo.email).toBe('customer@test.com')
        })
```

Reply:
I tried removing the act and using this instead but it produces the `not wrapped in act` error

Reply:
That is interesting, I use waitFor on my end, and the error did not show up. 🤔

Reply:
Whoa I figured it out. Combining the waitFor with the other change to use `jest.mock` instead of `jest.spyOn` removes the need to use `act`. 

Something about using `jest.spyOn` was requiring the `act` wrapper.